### PR TITLE
Added cultureSelector to dates formatting

### PR DESCRIPTION
**Pull request by savvinsergey**
Added cultureSelector to dates formattings
Update API to v1.5
Solve problem with dates values for unit tests

### CODE REVIEW

File: test/visualTest.ts [hunk 1]
```diff
@@ -233,6 +234,50 @@ module powerbi.extensibility.visual.test {
                 })(dateType));
             }
 
+            it("Verify date format for culture which user have chosen", (done) => {
+                let host = mocks.createVisualHost();
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Please specify a type

File: test/visualTest.ts [hunk 2]
```diff
@@ -233,6 +234,50 @@ module powerbi.extensibility.visual.test {
...
+                host.locale = host.locale || (<any>window.navigator).userLanguage || window.navigator["language"];
+                let dateFormatter = valueFormatter.create({format: "d", cultureSelector: host.locale});
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Please specify a type

File: test/visualTest.ts [hunk 3]
```diff
@@ -233,6 +234,50 @@ module powerbi.extensibility.visual.test {
...
+                    formattedDates.push(dateFormatter.format(date));
+                }
+
+                dataView = defaultDataViewBuilder.getDataView([
+                    GanttData.ColumnTask,
+                    GanttData.ColumnStartDate,
+                    GanttData.ColumnDuration]);
+
+                for (let dvColumn of dataView.metadata.columns) {
+                    if (!!dataView.categorical.categories) {
```
Comment (ignatvilesov) [inline test/visualTest.ts]: You shouldn't convert to boolean. JS converts it automatically

File: test/visualTest.ts [hunk 4]
```diff
@@ -233,6 +234,50 @@ module powerbi.extensibility.visual.test {
...
+                        for (let dvCategory of dataView.categorical.categories) {
+                            let roles = dvCategory.source.roles;
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Please specify a type

File: test/visualTest.ts [hunk 5]
```diff
@@ -233,6 +234,50 @@ module powerbi.extensibility.visual.test {
...
+                            if (roles && roles[GanttData.ColumnStartDate]) {
+                                dvColumn.format = "d";
+                            }
+                        }
+                    }
+                }
+
+                visualBuilder.updateRenderTimeout(dataView, () => {
+                    let tasks = d3.select(visualBuilder.element.get(0)).selectAll(".task").data();
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Please specify a type

File: test/visualTest.ts [hunk 6]
```diff
@@ -233,6 +234,50 @@ module powerbi.extensibility.visual.test {
...
+                    for (let task of tasks) {
+                        for (let tooltipInfo of task.tooltipInfo) {
+                            if (tooltipInfo.displayName === "Start Date") {
+                                let value  = tooltipInfo.value;
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Please specify a type

File: test/visualTest.ts [hunk 7]
```diff
@@ -233,6 +234,50 @@ module powerbi.extensibility.visual.test {
...
+                                let idx = formattedDates.indexOf(value);
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Please specify a type

File: test/visualTest.ts (hunk 1 above)
Comment (savvinsergey) [inline test/visualTest.ts]: Done

File: test/visualTest.ts (hunk 2 above)
Comment (savvinsergey) [inline test/visualTest.ts]: Done

File: test/visualTest.ts (hunk 3 above)
Comment (savvinsergey) [inline test/visualTest.ts]: Done

File: test/visualTest.ts (hunk 4 above)
Comment (savvinsergey) [inline test/visualTest.ts]: Done

File: test/visualTest.ts (hunk 5 above)
Comment (savvinsergey) [inline test/visualTest.ts]: Done

File: test/visualTest.ts (hunk 6 above)
Comment (savvinsergey) [inline test/visualTest.ts]: Done

File: test/visualTest.ts (hunk 7 above)
Comment (savvinsergey) [inline test/visualTest.ts]: Done

**Pull request merged**